nautobot/dcim/api/serializers.py

```diff
@@ -1025,6 +1025,16 @@ def validate(self, data):
                 )
 
         # Validate many-to-many VLAN assignments
+        mode = data.get("mode", getattr(self.instance, "mode", None))
+        has_tagged_vlans = data.get("tagged_vlans", False)
+        if not has_tagged_vlans and self.instance and self.instance.tagged_vlans.exists():
```

Comment:
Should this only raise an exception if trying to set `tagged_vlans` but not setting the mode to `tagged`? Should changing the mode from `tagged` to `access` and not specifying any tagged vlans also generate an exception? That's how it works in the UI.

Reply:
Only when mode is set to `tagged` is a tagged VLAN necessary; why should this generate an exception if mode is changed to 'access' and tagged VLANs are not added?

Reply:
Sorry my message was confusing. I think we should remove this part

```py
self.instance and self.instance.tagged_vlans.exists()
```

With this behavior you have to specify `mode: tagged` on every API call to an interface with `tagged_vlans` right?

Reply:
What happens if an interface has `tagged_vlans` and is `mode: tagged` and you want to change it to `mode: access`?

Reply:
The request in #2253, as I understand it, would be that you need to explicitly clear the tagged_vlans as part of the mode change.